### DISCUSSION TITLE
Is the size of the encrypted stream predictable?

### DISCUSSION THREAD

**Discussion by NickDarvey** [openpgpjs/openpgpjs#1382] [Q&A] [answered]
If I know the size of my `inputStream`, is it possible to predict the size of my `outputStream`? If so, how can I determine it?

```typescript
    const message = await openpgp.createMessage({ binary: inputStream });
    const outputStream = await openpgp.encrypt({
        message, // input as Message object
        passwords: ['secret stuff'], // multiple passwords possible
        format: 'binary' // don't ASCII armor (for Uint8Array output)
    });
```
(From the [README](https://github.com/openpgpjs/openpgpjs#streaming-encrypt-uint8array-data-with-a-password).)

**Comment by larabr** [comment, marked as answer]
Hi @NickDarvey , the output size depends on how the input stream was chunked, among other things, so I don't think there is a good way to predict this.

**Reply by NickDarvey** [reply]
My input stream is a [File](https://developer.mozilla.org/en-US/docs/Web/API/File) (i.e. `file.stream()`). Are File streams chunked? Or do you mean it depends on how OpenPGP chunks the input stream?

**Reply by larabr** [reply]
Yes, File streams are chunked -- chunking is necessary for streaming. For each chunk, we add some metadata when creating the OpenPGP message, which is why the output size depends on the input stream's chunking.

If you want to be able to easily compute the (streamed) encrypted data size, you might want to use an authenticated cipher directly (like AES-OCB or ChaChaPoly) instead of OpenPGP.

**Reply by NickDarvey** [reply]
Understood, thank you for elaborating

**Reply by NickDarvey** [reply]
Related: https://github.com/openpgpjs/web-stream-tools/issues/6#issuecomment-635596666
